dotcom-rendering/src/web/components/SignInGate/types.ts

```diff
@@ -18,15 +18,78 @@ export type SignInGateComponent = {
 	}: CanShowGateProps) => Promise<boolean>;
 };
 
+// Taking this approach for validating whether a specific
+// string is in a union type as per the advice of this blog:
+// https://dev.to/hansott/how-to-check-if-string-is-member-of-union-type-1j4m
+// Other option is to use an enum, though there are downside to this:
+// https://dev.to/azure/the-dangers-of-typescript-enums-55pd
+
+const ALL_USER_TYPES = ['new', 'guest', 'current'] as const;
```

Comment:
Doesn't seem to be a defacto approach for checking whether a string is a more specific type. Left some links as to how I got here, but open to suggestion on this.